Documentation/ExtensionArchitecture/HowTo/Localization/Php.rst

```diff
@@ -0,0 +1,92 @@
+.. include:: /Includes.rst.txt
+.. index::
+   Localization; PHP
+.. _extension-localization-php:
+
+====================
+Localization in PHP
+====================
+
+Sometimes you have to localize a string in the PHP code, for
+example inside of a controller or a ViewHelper.
+
+You can get the translation of a string into the current language via:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/SomeClass.php
+
+    $translatedString = $GLOBALS['LANG']->sL($translationKey);
+
+Where :php:`$translationKey` has the following format:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/SomeClass.php
+
+    use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+    $translationKey = 'LLL:EXT:' . GeneralUtility::camelCaseToLowerCaseUnderscored($extensionName)
```

Comment:
I find the usage of `camelCaseToLowerCaseUnderscored` overly complicated. In Extbase context I may have the extension name and can reuse it, mostly I have the extension key or just write it plain into the string.

If we go with this, we should link to the explanation of extension name/key, as many people don't know the difference, as I assume.

Reply:
Removed it